src/main/java/com/exasol/containers/workarounds/LogRotationWorkaround.java

```diff
@@ -0,0 +1,75 @@
+package com.exasol.containers.workarounds;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.Container.ExecResult;
+
+import com.exasol.containers.ExasolContainer;
+import com.exasol.containers.ExasolDockerImageReference;
+import com.exasol.containers.exec.ExitCode;
+
+/**
+ * This is a workaround for an issue with broken log rotation present in Exasol's `docker-db` version 7.0.x and below.
+ */
+public class LogRotationWorkaround implements Workaround {
+    static final Logger LOGGER = LoggerFactory.getLogger(LogRotationWorkaround.class);
+    @SuppressWarnings("java:S1075") // This is a fixed path. It won't change.
+    static final String ANACRON_SPOOL = "/var/spool/anacron/cron.daily";
+    private final ExasolContainer<? extends ExasolContainer<?>> exasol;
+
+    /**
+     * Create a new instance of a {@link LogRotationWorkaround}.
+     *
+     * @param exasol Exasol container where the workaround is applied
+     */
+    public LogRotationWorkaround(final ExasolContainer<? extends ExasolContainer<?>> exasol) {
+        this.exasol = exasol;
+    }
+
+    @Override
+    public String getName() {
+        return "log rotation";
+    }
+
+    // [impl->dsn~log-rotation-workaround-criteria~1]
+    @Override
+    public boolean isNecessary() {
+        if (this.exasol.isReused()) {
+            LOGGER.trace("Log rotation workaround unnecessary, since container is being reused.");
+            return false;
+        } else {
+            final ExasolDockerImageReference reference = this.exasol.getDockerImageReference();
+            if (reference.hasMajor() && //
+                    ((reference.getMajor() < 7) //
+                            || ((reference.getMajor() == 7) && reference.hasMinor() && (reference.getMinor() < 1))))
+
+            {
+                LOGGER.trace("Log rotation workaround required, since Exasol version is below 7.1.");
+                return true;
+            } else {
+                LOGGER.trace("Log rotation workaround unnecessary, since Exasol version is 7.1 or higher.");
+                return false;
+            }
+        }
+    }
+
+    // [impl->dsn~log-rotation-workaround~1]
+    @Override
+    public void apply() throws WorkaroundException {
```

Comment:
Different workarrounds could be applied at a different point in the container lifecycle. To make this API aware of that possibility, you could rename the method to `onContainerStart()` or something similar.

Reply:
Unfortunately that is not possible. We realistically need to have `execInContainer` available, so the Container must be initialized already. This limits the potential workaround, I know. But that's the way it is.